### PR TITLE
Refactor: undo changes to `RedBlackTree.foldRight`

### PR DESCRIPTION
**Pull request by ninaandrup**
Related to https://github.com/flix/flix/issues/3226

This PR includes all namespaces affected by `RedBlackTree`

- [x] `DelayMap`
- [x] `Map`
- [x] `MutMap`
- [x] `Set`
- [x] `MutSet`

### CODE REVIEW

File: main/src/library/Map.flix [hunk 1]
```diff
@@ -464,19 +464,38 @@ namespace Map {
...
+    pub def foldRightWithCont(f: (v, Unit -> b & ef) -> b & ef, z: b, m: Map[k, v]): b & ef =
+        foldRightWithKeyCont((_, v, c) -> f(v, c), z, m)
+
+    ///
+    /// Applies `f` to a start value `s` and all key-value pairs in `m` going from right to left.
+    ///
+    /// That is, the result is of the form: `f(k1, v1, ...f(kn-1, vn-1, f(kn, vn, s)))`.
+    /// A `foldRightWithKeyCont` allows early termination by not calling the continuation.
+    ///
+    pub def foldRightWithKeyCont(f: (k, v, Unit -> b & ef) -> b & ef, z: b, m: Map[k, v]): b & ef =
```
Comment (ninaandrup) [inline main/src/library/Map.flix:495]: @magnus-madsen I'm not sure, if this name `Map.foldrightWithKeyCont` is good? It is the `Map.foldRightWithKey` that calls `RedBlackTree.foldRightWithCont`.

File: main/src/library/RedBlackTree.flix [hunk 2]
```diff
@@ -289,14 +289,23 @@ namespace RedBlackTree {
...
     ///
     /// Applies `f` to a start value `z` and all key-value pairs in `t` going from right to left.
     ///
     /// That is, the result is of the form: `f(k1, v1, ...f(kn-1, vn-1, f(kn, vn, s)))`.
-    /// A `foldRight` allows early termination by not calling the continuation.
+    /// A `foldRightWithCont` allows early termination by not calling the continuation.
     ///
-    pub def foldRight(f: (k, v, Unit -> b & ef) -> b & ef, z: b, t: RedBlackTree[k, v]): b & ef = match t {
-        case Node(_, a, k, v, b) => foldRight(f, f(k, v, _ -> foldRight(f, z, b)), a)
+    pub def foldRightWithCont(f: (k, v, Unit -> b & ef) -> b & ef, z: b, tree: RedBlackTree[k, v]): b & ef = match tree {
```
Comment (ninaandrup) [inline main/src/library/RedBlackTree.flix:307]: I have noticed that some of the functions in `RedBlackTree` uses `tree` and some uses `t`. Should this be consistent? Only 5 uses `tree` and the rest `t`.

File: main/src/library/Map.flix (hunk 1 above)
Comment (magnus-madsen) [reply main/src/library/Map.flix:493]: I think its OK.

File: main/src/library/RedBlackTree.flix (hunk 2 above)
Comment (magnus-madsen) [reply main/src/library/RedBlackTree.flix:307]: Yes, would be nice. Probably just use `t`.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**